### DISCUSSION TITLE
Going crazy over portainer, mqtt and zigbee2mqtt - please help

### DISCUSSION THREAD

**Discussion by remirr** [Koenkk/zigbee2mqtt#19135] [General]
Ok so I have a raspberry pi and bought a bunch of Tradfri Zigbee devices from Ikea to try and control power plugs through HA. I have other things running on the Pi and so I can't install HA as an 'OS' which also means that I can't use add-ons, hence using MQTT + Docker + Zigbee2MQTT + Portainer. 

This is where the problem comes in, I spent 5 hours trying to get Zigbee2MQTT to run under Docker, I couldn't get that to work, then I tried Docker-Compose, I couldn't get that to work either, finally I found the tool Portainer (which I didn't know existed), followed an old guide but am still having touble, yesterday I was seeing somthing like:

**Waiting for 177.17.0.1**

**Now I am seeing this:**
`Zigbee2MQTT:info  2023-10-01 17:42:12: Logging to console and directory: '/app/data/log/2023-10-01.17-42-12' filename: log.txt
Zigbee2MQTT:info  2023-10-01 17:42:12: Starting Zigbee2MQTT version 1.33.0 (commit #7ee207f)
Zigbee2MQTT:info  2023-10-01 17:42:12: Starting zigbee-herdsman (0.18.5)
Zigbee2MQTT:info  2023-10-01 17:42:18: zigbee-herdsman started (resumed)
Zigbee2MQTT:info  2023-10-01 17:42:18: Coordinator firmware version: '{"meta":{"maintrel":"3 ","majorrel":"6","minorrel":"10","product":8,"revision":"6.10.3.0 build 297"},"type":"EZSP v8"}'
Zigbee2MQTT:info  2023-10-01 17:42:18: Currently 0 devices are joined:
Zigbee2MQTT:warn  2023-10-01 17:42:18: `permit_join` set to  `true` in configuration.yaml.
Zigbee2MQTT:warn  2023-10-01 17:42:18: Allowing new devices to join.
Zigbee2MQTT:warn  2023-10-01 17:42:18: Set `permit_join` to `false` once you joined all devices.
Zigbee2MQTT:info  2023-10-01 17:42:18: Zigbee: allowing new devices to join.
Zigbee2MQTT:info  2023-10-01 17:42:18: Connecting to MQTT server at mqtt://172.17.0.1
Zigbee2MQTT:error 2023-10-01 17:42:18: MQTT error: connect ECONNREFUSED 172.17.0.1:1883
Zigbee2MQTT:error 2023-10-01 17:42:18: MQTT failed to connect, exiting...
Zigbee2MQTT:info  2023-10-01 17:42:18: Stopping zigbee-herdsman...
Zigbee2MQTT:info  2023-10-01 17:42:18: Stopped zigbee-herdsman
Using '/app/data' as data directory`

So then I realised that Zigbee2MQTT didn't include an MQTT server, so then I spent the next day trying to get MQTT to run but cannot for the life of me get it to run. I assume it can't connect to MQTT because I am assuming its not installed. I have been trying to install MQTT using portainer for 2 days now and I just keep getting errors of the container not bing able to launch or that you can't have a bridge network or that the port 1883 is already taken etc.

Is there a simple guide anyone has on installing HA + MQTT with Portainer? All I want to do is get Zigbee to work with my Sonoff stick so that I can connect these devices. 

Thank you very much

**Comment by francisp2** [comment]
Why struggle with docker ? 
sudo apt-get mosquitto
and you are done.

**Reply by remirr** [reply]
But it is not that easy, you need to add a bunch of stuff to docker-compose.yml and then also the mosquitto.conf file. Some websites say you need to create authentication, others say you can allow anonymous

**Reply by remirr** [reply]
So I just removed mqtt and tried your method. It installed and showed this after but has not used any of the .conf files etc.. I have put in docker. When I start Zigbee2MQTT I get this error. 

`Using '/app/data' as data directory
Zigbee2MQTT:info  2023-10-01 20:45:57: Logging to console and directory: '/app/data/log/2023-10-01.20-45-57' filename: log.txt
Zigbee2MQTT:info  2023-10-01 20:45:57: Starting Zigbee2MQTT version 1.33.0 (commit #7ee207f)
Zigbee2MQTT:info  2023-10-01 20:45:57: Starting zigbee-herdsman (0.18.5)
Zigbee2MQTT:info  2023-10-01 20:46:03: zigbee-herdsman started (resumed)
Zigbee2MQTT:info  2023-10-01 20:46:03: Coordinator firmware version: '{"meta":{"maintrel":"3 ","majorrel":"6","minorrel":"10","product":8,"revision":"6.10.3.0 build 297"},"type":"EZSP v8"}'
Zigbee2MQTT:info  2023-10-01 20:46:03: Currently 0 devices are joined:
Zigbee2MQTT:warn  2023-10-01 20:46:03: `permit_join` set to  `true` in configuration.yaml.
Zigbee2MQTT:warn  2023-10-01 20:46:03: Allowing new devices to join.
Zigbee2MQTT:warn  2023-10-01 20:46:03: Set `permit_join` to `false` once you joined all devices.
Zigbee2MQTT:info  2023-10-01 20:46:03: Zigbee: allowing new devices to join.
Zigbee2MQTT:info  2023-10-01 20:46:03: Connecting to MQTT server at mqtt://172.17.0.1
Zigbee2MQTT:error 2023-10-01 20:46:03: MQTT error: connect ECONNREFUSED 172.17.0.1:1883
Zigbee2MQTT:error 2023-10-01 20:46:03: MQTT failed to connect, exiting...
Zigbee2MQTT:info  2023-10-01 20:46:03: Stopping zigbee-herdsman...
Zigbee2MQTT:info  2023-10-01 20:46:04: Stopped zigbee-herdsman
Using '/app/data' as data directory
Zigbee2MQTT:info  2023-10-01 20:46:10: Logging to console and directory: '/app/data/log/2023-10-01.20-46-10' filename: log.txt
Zigbee2MQTT:info  2023-10-01 20:46:10: Starting Zigbee2MQTT version 1.33.0 (commit #7ee207f)
Zigbee2MQTT:info  2023-10-01 20:46:10: Starting zigbee-herdsman (0.18.5)
Zigbee2MQTT:error 2023-10-01 20:46:11: Not connected to MQTT server!
Zigbee2MQTT:error 2023-10-01 20:46:11: Cannot send message: topic: 'zigbee2mqtt/bridge/state', payload: 'offline
Zigbee2MQTT:info  2023-10-01 20:46:11: Disconnecting from MQTT server
TypeError: Cannot read properties of undefined (reading 'end')
    at MQTT.disconnect (/app/lib/mqtt.ts:130:21)
    at Controller.stop (/app/lib/controller.ts:198:9)
    at stop (/app/index.js:111:5)
    at process.handleQuit (/app/index.js:117:9)`

When I installed MQTT using apt-get I got this. 

https://tinypic.host/image/8hOcw

**Comment by francisp2** [comment]
Mosquitto 2..0.2 and above

Mosquitto v2 introduced some important changes that affect first time users in particular.

By default it requires authentication and doesn’t listen on a network address.

The following simple configuration file will make mosquitto start like previous versions:

so in /etc/mosquitto/mosquitto.conf add 

listener 1883
allow_anonymous true
